### PR TITLE
feat: add --default-agent-pod-labels flag for global agent pod labels

### PR DESCRIPTION
**Pull request by EItanya**
## Summary
- Adds a new `--default-agent-pod-labels` CLI flag / `DEFAULT_AGENT_POD_LABELS` env var that applies a set of labels to all agent pod templates created by kagent
- Labels are specified as comma-separated `key=value` pairs (e.g. `team=platform,environment=production`)
- Fully wired through Helm via `controller.agentDeployment.podLabels` map in values.yaml
- Label precedence: built-in defaults → global default labels → per-agent `spec.deployment.labels`

## Test plan
- [x] Unit tests for `ParseLabels` (8 cases covering valid input, edge cases, and errors)
- [x] Unit test for `DEFAULT_AGENT_POD_LABELS` env var loading
- [x] Existing translator golden tests pass
- [x] 4 new Helm unit tests (labels set, empty, not configured, single label)
- [x] Deployed to Kind cluster with `podLabels: {team: platform, environment: production}` — verified all 11 agent pod templates contain both labels
- [x] `kubectl get pods -l team=platform` returns all agent pods

Comment left by Claude on behalf of @EItanya

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: go/core/pkg/app/app_test.go [hunk 1]
```diff
@@ -265,6 +265,92 @@ func TestDatabaseUrlFileFlag(t *testing.T) {
 	assert.Equal(t, "/etc/credentials/db-url", cfg.Database.UrlFile)
 }
 
+func TestMapValue(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   string
+		want    map[string]string
+		wantErr bool
+	}{
```
Comment (iplay88keys) [inline go/core/pkg/app/app_test.go:274]: Missing the test case where an env var contains an equals (such as: `key=val=ue`). This could occur with a base64 value, for example, so we should have a test case for it.

File: go/core/pkg/app/app_test.go [hunk 2]
```diff
@@ -265,6 +265,92 @@ func TestDatabaseUrlFileFlag(t *testing.T) {
...
+	}
+}
+
+func TestMapValueString(t *testing.T) {
+	var target map[string]string
+	mv := &MapValue{Target: &target}
+	assert.Equal(t, "", mv.String())
+
+	target = map[string]string{"team": "platform"}
+	assert.Equal(t, "team=platform", mv.String())
```
Comment (iplay88keys) [inline go/core/pkg/app/app_test.go:344]: Worth noting that iterating over a map in Go is random, so if there were multiple keys here, the output would be inconsistent. It's worth considering sorting before ranging over the keys in the implementation.
